### DISCUSSION TITLE
"enabled:" not working on anchor elements unless i use plugin

### DISCUSSION THREAD

**Discussion by jaxramus** [tailwindlabs/tailwindcss#8668] [Help]
I'm trying to use `enabled:` on anchor elements

it works fine on button elements

the only way i can get it to work on anchor elements is to add this to my plugins:

    plugins: [
        plugin(function ({ addVariant }) {
            addVariant('enabled', '&:not(:disabled)')
        })
    ],

is there a way to make `enabled:` work with non-buttons without having to use the `addVariant` plugin?

**Comment by rohmann** [comment]
Hi @jaxramus,

This has to do with a quirk in how HTML/CSS handles `:enabled` `:disabled`. First off, technically they're not supported on anchor tags. Notice that when `disabled` is present the browser behavior doesn't change. But, for some reason doing `:not(:disabled)` is still honored - it even works on divs. The Tailwind variants utilize the native `:enabled` pseudo selector which is technically unsupported on anchor tags.

To be safe, I'd probably make a new variant that uses an attribute selector instead. This is a bit more heavy handed and works around the browser limiations, but will allow creating styles on any element that apply in the absence of a `disabled` attribute. Here's an example

https://play.tailwindcss.com/XhkPhaG8zb
